src/units/attack_type.hpp

```diff
@@ -132,8 +132,9 @@ class attack_type : public std::enable_shared_from_this<attack_type>
 	 * @param temp_list the list checked and returned.
 	 * @param abil_list list checked for verify presence of overwrite_specials .
 	 * @param filter_self name of [filter_"self/student"] if is abilities or specials who are checked
+	 * @param is_special for determine if list is a special or a ability.
 	 */
-	unit_ability_list overwrite_special_checking(const std::string& ability, unit_ability_list temp_list, const unit_ability_list& abil_list, const std::string& filter_self = "filter_self") const;
+	unit_ability_list overwrite_special_checking(const std::string& ability, const unit_ability_list& temp_list, const unit_ability_list& abil_list, const std::string& filter_self = "filter_student", bool is_special = false) const;
```

Comment:
Instead of changing a default value from `filter_self` to `filter_student`, please just remove the default so that all callers have to provide the value. Without this, someone comparing code between 1.16 and 1.17 may incorrectly assume that `overwrite_special_checking(special, spe_list, abil_list);` has the same behavior in both branches; that's a gotcha that's easy to avoid.

Reply:
it's done